src/components/Header/Header.jsx

```diff
@@ -14,16 +15,48 @@ import SeachIcon from '../../ui/icons/Search';
 
 import logo from '../../assets/img/wizeline-logo.png';
 
-const Header = () => {
+import { youtubeGetVideos, youtubeSearchVideos } from '../../state/actions/youtube';
+import { useYoutubeVideo } from '../../state/Provider';
+
+export const retrieveVideosEffect = ({ search, dispatch, hideSearcher }) => {
+  if (hideSearcher) return;
+
+  const retriveVideosFetch = async () => {
+    if (search && search.trim()) {
+      dispatch(await youtubeSearchVideos(search));
+    } else {
+      dispatch(await youtubeGetVideos());
+    }
+  };
+  retriveVideosFetch();
+};
+
+const Header = ({ hideSearcher }) => {
+  const history = useHistory();
+  const [search, setSearch] = useState('');
+  const { dispatch } = useYoutubeVideo();
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => retrieveVideosEffect({ search, dispatch, hideSearcher }), [search]);
```

Comment:
You should add all the variables that you use for you effect as dependencies like this:

```suggestion
  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => retrieveVideosEffect({ search, dispatch, hideSearcher }), [search, dispatch, hideSearcher]);
```